src/adapter/src/coord/ddl.rs

```diff
@@ -857,17 +858,22 @@ impl Coordinator {
     }
 
     async fn drop_vpc_endpoints(&mut self, vpc_endpoints: Vec<GlobalId>) {
+        let cp = self.cloud_resource_controller
+            .as_ref()
+            .ok_or(AdapterError::Unsupported("AWS PrivateLink connections"))
+            .expect("vpc endpoints should only be dropped in CLOUD, where `cloud_resource_controller` is `Some`");
         for vpc_endpoint in vpc_endpoints {
-            if let Err(e) = self
-                .cloud_resource_controller
-                .as_ref()
-                .ok_or(AdapterError::Unsupported("AWS PrivateLink connections"))
-                .expect("vpc endpoints should only be dropped in CLOUD, where `cloud_resource_controller` is `Some`")
-                .delete_vpc_endpoint(vpc_endpoint)
+            if let Err(e) = cp
+                .delete_vpc_endpoint_in_task_if(InTask::Yes, vpc_endpoint)
+                .instrument(info_span!(
+                    "coord::catalog_transact_with_side_effects::drop_vpc_endpoint"
+                ))
                 .await
             {
-                warn!("Dropping VPC Endpoints has encountered an error: {}", e);
-                // TODO reschedule this https://github.com/MaterializeInc/cloud/issues/4407
+                {
+                    warn!("Dropping VPC Endpoints has encountered an error: {}", e);
+                    // TODO reschedule this https://github.com/MaterializeInc/cloud/issues/4407
+                }
```

Comment:
I think the `_in_task_if` pattern is probably not the right thing to follow here! That infrastructure exists for the storage stuff because we need the ability to toggle between running in a task or not. In this case we don't have the same conditional requirements.

I think, since we have no need to drop VPC endpoints in the foreground, in this case it'd be simpler to just call this function `drop_vpc_endpoints_in_background` and spawn the task here, rather than making it part of the cloud resources controller.